File: bin/src/hadd.rs
use std::io::{self, BufRead, Write};

fn main() -> io::Result<()> {
    let stdin = io::stdin();
    let mut stdout = io::stdout();

    for raw_line in stdin.lock().lines() {
        let line = raw_line?;
        let trimmed = line.trim();

        if trimmed.is_empty() {
            // The Python version emits a lone marker when callers send spacing lines, so we preserve that contract verbatim.
            writeln!(stdout, "#")?;
        } else {
            // We intentionally stack comment markers so repeated passes through this tool make the accumulated quoting explicit and reversible via commremove.
            writeln!(stdout, "# {trimmed}")?;
        }
    }

    stdout.flush()?; // Flushing intentionally mirrors the python script's immediate stdout behavior so pipelines never block on buffered data.
    Ok(())
}

File: bin/src/hadd.go
//go:build ignore

package main

import (
	"bufio"
	"fmt"
	"os"
	"strings"
)

func main() {
	scanner := bufio.NewScanner(os.Stdin)

	for scanner.Scan() {
		line := scanner.Text()
		trimmed := strings.TrimSpace(line)

		if trimmed == "" {
			// The Python version emits a lone marker when callers send spacing lines, so we preserve that contract verbatim.
			fmt.Println("#")
		} else {
			// We intentionally stack comment markers so repeated passes through this tool make the accumulated quoting explicit and reversible via commremove.
			fmt.Printf("# %s\n", trimmed)
		}
	}

	if err := scanner.Err(); err != nil {
		fmt.Fprintf(os.Stderr, "error reading input: %v\n", err)
		os.Exit(1)
	}
}
